arroyo/processing/processor.py

```diff
@@ -394,39 +392,44 @@ def _run_once(self) -> None:
                     # to pause the consumer and hold the message until it is
                     # accepted, at which point we can resume consuming.
                     if not message_carried_over:
-                        logger.debug(
-                            "Caught %r while submitting %r, pausing consumer...",
-                            e,
-                            self.__message,
-                        )
-                        self.__consumer.pause([*self.__consumer.tell().keys()])
+                        if self.__backpressure_timestamp is None:
+                            self.__backpressure_timestamp = time.time()
+
+                            if not self.__is_paused and (
+                                time.time() - self.__backpressure_timestamp > 1
+                            ):
+                                logger.debug(
+                                    "Caught %r while submitting %r, pausing consumer...",
+                                    e,
+                                    self.__message,
+                                )
+                                self.__consumer.pause([*self.__consumer.tell().keys()])
+                                self.__is_paused = True
 
-                        self.__paused_timestamp = time.time()
                     else:
-                        current_time = time.time()
                         time.sleep(0.01)
-                        if self.__paused_timestamp:
-                            self.__metrics_buffer.incr_timing(
-                                "arroyo.consumer.paused.time",
-                                current_time - self.__paused_timestamp,
-                            )
```

Comment:
Is there a way to still capture the consumer pause time? If we lose visibility into consumer pauses, troubleshooting consumer bottlenecks becomes hard.

Reply:
Oh, we're actually still recording the paused time with the same metric if you look below. I just removed this contraption that re-records on every iteration of the loop and then updates the timestamp because i think that's overkill and makes the code more complicated. Now we just record it once when we unpause, which is what we ended up doing in the Rust code too.